### PR TITLE
STYLE enable ruff PLR5501 #51709

### LINKED ISSUE #51709: STYLE enable ruff PLR5501
Task is:

1. check the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing_environment.html) for how to set up your environment
2. remove `PLR5501` from

https://github.com/pandas-dev/pandas/blob/4d18871d398248c5876035f5aa139292e6a05279/pyproject.toml#L274-L275

3. run `pre-commit run ruff --all-files`
4. fixup the errors it flags
5. stage, commit, push, open pull request

### PR DESCRIPTION
**Pull request by ahmad-04**
- [x] closes #51709
- [x] Added "elif instead of else if"

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
well done! looks like there's a merge conflict, could you resolve it please?

**Comment by ahmad2901** [comment]
Hey, merge conflict is this one I wrote below but idk how to fix this.


<<<<<<< shiny-new-feature
# to be enabled gradually
"pandas/core/*" = ["PLR5501"]
=======
# TCH to be enabled gradually
"pandas/core/arrays/*" = ["TCH"]
"pandas/core/io/*" = ["TCH"]
"pandas/core/indexers/*" = ["TCH"]
"pandas/core/indexes/*" = ["TCH"]
"pandas/core/internals/*" = ["TCH"]
"pandas/core/groupby/*" = ["TCH"]
"pandas/core/methods/*" = ["TCH"]
"pandas/core/array_algos/*" = ["TCH"]
"pandas/core/dtypes/*" = ["TCH"]
"pandas/core/generic.py" = ["TCH"]
"pandas/core/frame.py" = ["TCH"]
"pandas/core/series.py" = ["TCH"]
"pandas/core/resample.py" = ["TCH"]
"pandas/core/nanops.py" = ["TCH"]
"pandas/core/apply.py" = ["TCH"]
"pandas/core/base.py" = ["TCH"]
"pandas/core/algorithms.py" = ["TCH"]
"pandas/core/ops/*" = ["TCH"]
"pandas/core/sorting.py" = ["TCH"]
"pandas/core/construction.py" = ["TCH"]
"pandas/core/common.py" = ["TCH"]
"pandas/core/missing.py" = ["TCH"]
"pandas/core/util/*" = ["TCH"]
"pandas/core/reshape/*" = ["TCH"]
"pandas/core/strings/*" = ["TCH"]
"pandas/core/tools/*" = ["TCH"]
"pandas/core/window/*" = ["TCH"]
"pandas/io/*" = ["TCH"]
"pandas/tseries/*" = ["TCH"]
"pandas/tests/*" = ["TCH"]
"pandas/plotting/*" = ["TCH"]
"pandas/util/*" = ["TCH"]
"pandas/_libs/*" = ["TCH"]
# Keep this one enabled
"pandas/_typing.py" = ["TCH"]
>>>>>>> main

**Comment by MarcoGorelli** [comment]
hey - please check https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts

File: pandas/io/parsers/python_parser.py [hunk 1]
```diff
@@ -555,20 +555,18 @@ def _infer_columns(
...
+                num_original_columns = len(names)
+            elif not callable(self.usecols) and len(names) != len(self.usecols):
+                raise ValueError(
+                    "Number of passed names did not match number of "
+                    "header fields in the file"
+                )
+            # Ignore output but set used columns.
+            self._handle_usecols([names], names, ncols)
+            columns = [names]
+            num_original_columns = ncols
```
Comment (MarcoGorelli) [inline pandas/io/parsers/python_parser.py:569]: does this not have to be in an `else` block?

Comment (ahmad2901) [reply pandas/io/parsers/python_parser.py:569]: Let me check

Comment (MarcoGorelli) [reply pandas/io/parsers/python_parser.py:569]: thanks

also, please make sure to run `pre-commit run --all-files`, as some other hooks are erroring

File: pandas/io/parsers/python_parser.py [hunk 2]
```diff
@@ -555,20 +555,18 @@ def _infer_columns(
...
-                    # Ignore output but set used columns.
-                    self._handle_usecols([names], names, ncols)
-                    columns = [names]
-                    num_original_columns = ncols
+            elif self.usecols is None or len(names) >= num_original_columns:
+                columns = self._handle_usecols([names], names, num_original_columns)
+                num_original_columns = len(names)
+                if not callable(self.usecols) and len(names) != len(self.usecols):
+                    raise ValueError(
+                        "Number of passed names did not match number of "
```
Comment (MarcoGorelli) [inline pandas/io/parsers/python_parser.py:563]: does this still need to be in the else block?

Comment (ahmad2901) [reply pandas/io/parsers/python_parser.py:563]: it was before changing to elif.

Comment (ahmad2901) [reply pandas/io/parsers/python_parser.py]: How did you know error was on that specific line???

File: doc/source/whatsnew/v2.1.0.rst [hunk 3]
```diff
@@ -92,8 +92,8 @@ Other API changes
 
 Deprecations
 ~~~~~~~~~~~~
-- Deprecated accepting slices in :meth:`DataFrame.take`, call ``obj[slicer]`` or pass a sequence of integers instead (:issue:`51539`)
 - Deprecating pinning ``group.name`` to each group in :meth:`SeriesGroupBy.aggregate` aggregations; if your operation requires utilizing the groupby keys, iterate over the groupby object instead (:issue:`41090`)
+- Deprecated accepting slices in :meth:`DataFrame.take`, call ``obj[slicer]`` or pass a sequence of integers instead (:issue:`51539`)
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v2.1.0.rst:96]: this shouldn't have changed, I presume something didn't go quite right when merging? could you restore it please?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
something's not quite right, tests are failing - please check the logic of each change

File: doc/source/whatsnew/v2.1.0.rst (hunk 3 above)
Comment (ahmad2901) [reply doc/source/whatsnew/v2.1.0.rst:96]: Hard reset before merge???

File: pandas/io/parsers/python_parser.py (hunk 1 above)
Comment (MarcoGorelli) [inline pandas/io/parsers/python_parser.py:569]: this needs to be in an else block

**Review by ahmad-04 — COMMENTED** [review]
I checked all changes. Did change one u suggested other than that there were not any that i found.

**Comment by ahmad2901** [comment]
Hey, this new error is because of numpy i think. it is not being import for some reason.

**Comment by MarcoGorelli** [comment]
yup, that one's unrelated - this should be good to go, I'll check again later, thanks!

**Comment by ahmad2901** [comment]
Hey @MarcoGorelli , did u find anything?

File: pandas/io/formats/format.py [hunk 4]
```diff
@@ -2103,11 +2103,11 @@ def __call__(self, num: float) -> str:
 
         if self.use_eng_prefix:
             prefix = self.ENG_PREFIXES[int_pow10]
+
```
Comment (MarcoGorelli) [inline pandas/io/formats/format.py:2106]: this is really minor, but could you please undo these unrelated newline changes? there's a few of them across the files you've modified, please check through them

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by ahmad2901** [comment]
Hey,

**Comment by ahmad2901** [comment]
Hey,
Thanks for advice and helping solve these issue checks cleared. what should I do next?

**Comment by MarcoGorelli** [comment]
nothing, this looks ready, feel free to take on other issues if interested - I'll merge when I've had a chance to double-check everything

**Review by MarcoGorelli — APPROVED** [review]
very nice, thanks @ahmad2901 !

no need to do anything, I'll merge once the final jobs are green

**Pull request merged**